<h1>UK Road Accidents</h1>
<hr>
<h3>Tanay, Xynex Kim R.</h3>
<h3>BSIT 3A</h3>

In [1]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

In [2]:
uk_accidents = pd.read_csv("datasets//uk_road_accident.csv")

In [3]:
uk_accidents

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious      5/6/2019  51.506187   
1       200701BS65737           Serious      2/7/2019  51.495029   
2       200701BS66127           Serious    26-08-2019  51.517715   
3       200701BS66128           Serious    16-08-2019  51.495478   
4       200701BS66837            Slight      3/9/2019  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight    18-02-2022  57.374005   
660675  201091NM01881            Slight    21-02-2022  57.232273   
660676  201091NM01935            Slight    23-02-2022  57.585044   
660677  201091NM01964           Serious    23-02-2022  57.214898   
660678  201091NM02142           Serious    28-02-2022  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
4             Darkness - lights lit  Kensington and Chelsea  -0.192487   
...                             ...                     ...        ...   
660674                     Daylight                Highland  -3.467828   
660675       Darkness - no lighting                Highland  -3.809281   
660676                     Daylight                Highland  -3.862727   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2                      NaN               Urban                    NaN   
3       Single carriageway               Urban     Fine no high winds   
4                      NaN               Urban                    NaN   
...                    ...                 ...                    ...   
660674  Single carriageway               Rural     Fine no high winds   
660675  Single carriageway               Rural     Fine no high winds   
660676  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type  
0                                        Car  
1                                        Car  
2                      Taxi/Private hire car  
3       Bus or coach (17 or more pass seats)  
4                              Other vehicle  
...                                      ...  
660674                                   Car  
660675                                   Car  
660676                                   Car  
660677                 Motorcycle over 500cc  
660

In [4]:
uk_accidents.describe()

Latitude      Longitude  Number_of_Casualties  Number_of_Vehicles
count  660654.000000  660653.000000         660679.000000       660679.000000
mean       52.553866      -1.431210              1.357040            1.831255
std         1.406922       1.383330              0.824847            0.715269
min        49.914430      -7.516225              1.000000            1.000000
25%        51.490690      -2.332291              1.000000            1.000000
50%        52.315641      -1.411667              1.000000            2.000000
75%        53.453452      -0.232869              1.000000            2.000000
max        60.757544       1.762010             68.000000           32.000000

In [5]:
uk_accidents.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 660679 entries, 0 to 660678
Data columns (total 14 columns):
 #   Column                   Non-Null Count   Dtype  
---  ------                   --------------   -----  
 0   Index                    660679 non-null  object 
 1   Accident_Severity        660679 non-null  object 
 2   Accident Date            660679 non-null  object 
 3   Latitude                 660654 non-null  float64
 4   Light_Conditions         660679 non-null  object 
 5   District Area            660679 non-null  object 
 6   Longitude                660653 non-null  float64
 7   Number_of_Casualties     660679 non-null  int64  
 8   Number_of_Vehicles       660679 non-null  int64  
 9   Road_Surface_Conditions  659953 non-null  object 
 10  Road_Type                656159 non-null  object 
 11  Urban_or_Rural_Area      660664 non-null  object 
 12  Weather_Conditions       646551 non-null  object 
 13  Vehicle_Type             660679 non-null  object 
dtypes: f

<h2>Filling null values.</h2>

In [6]:
uk_accidents.isnull()

Index  Accident_Severity  Accident Date  Latitude  Light_Conditions  \
0       False              False          False     False             False   
1       False              False          False     False             False   
2       False              False          False     False             False   
3       False              False          False     False             False   
4       False              False          False     False             False   
...       ...                ...            ...       ...               ...   
660674  False              False          False     False             False   
660675  False              False          False     False             False   
660676  False              False          False     False             False   
660677  False              False          False     False             False   
660678  False              False          False     False             False   

        District Area  Longitude  Number_of_Casualties  Number_of_Vehicles  \
0               False      False                 False               False   
1               False      False                 False               False   
2               False      False                 False               False   
3               False      False                 False               False   
4               False      False                 False               False   
...               ...        ...                   ...                 ...   
660674          False      False                 False               False   
660675          False      False                 False               False   
660676          False      False                 False               False   
660677          False      False                 False               False   
660678          False      False                 False               False   

        Road_Surface_Conditions  Road_Type  Urban_or_Rural_Area  \
0                         False      False                False   
1                         False      False                False   
2                         False       True                False   
3                         False      False                False   
4                         False       True                False   
...                         ...        ...                  ...   
660674                    False      False                False   
660675                    False      False                False   
660676                    False      False                False   
660677                    False      False                False   
660678                    False      False                False   

        Weather_Conditions  Vehicle_Type  
0                    False         False  
1                    False         False  
2                     True         False  
3                    False         False  
4                     True         False  
...                    ...           ...  
660674               False         False  
660675               False         False  
660676               False         False  
660677               False         False  
660678               False         False  

[660679 rows x 14 columns]

In [7]:
uk_accidents.isnull().sum()

Index                          0
Accident_Severity              0
Accident Date                  0
Latitude                      25
Light_Conditions               0
District Area                  0
Longitude                     26
Number_of_Casualties           0
Number_of_Vehicles             0
Road_Surface_Conditions      726
Road_Type                   4520
Urban_or_Rural_Area           15
Weather_Conditions         14128
Vehicle_Type                   0
dtype: int64

In [8]:
uk_accidents['Latitude'].mean()

np.float64(52.553865761110956)

In [9]:
uk_accidents['Road_Surface_Conditions'].mode()

0    Dry
Name: Road_Surface_Conditions, dtype: object

In [10]:
uk_accidents['Road_Type'].mode()

0    Single carriageway
Name: Road_Type, dtype: object

In [11]:
uk_accidents['Urban_or_Rural_Area'].mode()

0    Urban
Name: Urban_or_Rural_Area, dtype: object

In [12]:
uk_accidents['Weather_Conditions'].mode()

0    Fine no high winds
Name: Weather_Conditions, dtype: object

In [13]:
uk_accidents['Longitude'].fillna(uk_accidents['Longitude'].mean())

0        -0.209082
1        -0.173647
2        -0.210215
3        -0.202731
4        -0.192487
            ...   
660674   -3.467828
660675   -3.809281
660676   -3.862727
660677   -3.823997
660678   -3.895673
Name: Longitude, Length: 660679, dtype: float64

In [14]:
uk_accidents['Latitude'] = uk_accidents['Latitude'].fillna(uk_accidents['Latitude'].mean())
uk_accidents['Longitude'] = uk_accidents['Longitude'].fillna(uk_accidents['Longitude'].mean())
uk_accidents['Road_Surface_Conditions'] = uk_accidents['Road_Surface_Conditions'].fillna(uk_accidents['Road_Surface_Conditions'].mode()[0])
uk_accidents['Road_Type'] = uk_accidents['Road_Type'].fillna(uk_accidents['Road_Type'].mode()[0])
uk_accidents['Urban_or_Rural_Area'] = uk_accidents['Urban_or_Rural_Area'].fillna(uk_accidents['Urban_or_Rural_Area'].mode()[0])
uk_accidents['Weather_Conditions'] = uk_accidents['Weather_Conditions'].fillna(uk_accidents['Weather_Conditions'].mode()[0])

In [15]:
uk_accidents.isnull().sum()

Index                      0
Accident_Severity          0
Accident Date              0
Latitude                   0
Light_Conditions           0
District Area              0
Longitude                  0
Number_of_Casualties       0
Number_of_Vehicles         0
Road_Surface_Conditions    0
Road_Type                  0
Urban_or_Rural_Area        0
Weather_Conditions         0
Vehicle_Type               0
dtype: int64

<hr>

In [16]:
from scipy.stats import f_oneway

<hr>

<h2>01. What's the total count of accident reports in the dataset?</h2>

In [17]:
uk_accidents["Index"].value_counts()

Index
2.01E+12         239478
2.01E+86              4
2.01E+197             4
2.01E+98              4
2.01E+175             4
                  ...  
201091NM01760         1
201091NM01881         1
201091NM01935         1
201091NM01964         1
201091NM02142         1
Name: count, Length: 421020, dtype: int64

<h3>Insight: According to the index, the CSV file holds 421,020 accident entries in total.</h3>

<hr>

<h2>02. How many distrct areas are there in the dataset?</h2>

In [18]:
uk_accidents["District Area"].unique()

array(['Kensington and Chelsea', 'Westminster', 'Richmond upon Thames',
       'Hammersmith and Fulham', 'Hounslow', 'Tower Hamlets',
       'City of London', 'Southwark', 'Camden', 'Hackney', 'Islington',
       'Barnet', 'Brent', 'Haringey', 'Merton', 'Ealing', 'Enfield',
       'Greenwich', 'Newham', 'London Airport (Heathrow)', 'Hillingdon',
       'Waltham Forest', 'Redbridge', 'Barking and Dagenham', 'Bromley',
       'Havering', 'Croydon', 'Lambeth', 'Wandsworth', 'Sutton', 'Bexley',
       'Lewisham', 'Harrow', 'Kingston upon Thames', 'Barrow-in-Furness',
       'South Lakeland', 'Carlisle', 'Eden', 'Allerdale', 'Copeland',
       'Fylde', 'Blackpool', 'Wyre', 'Lancaster', 'Chorley',
       'West Lancashire', 'South Ribble', 'Preston',
       'Blackburn with Darwen', 'Hyndburn', 'Ribble Valley', 'Burnley',
       'Pendle', 'Rossendale', 'Wirral', 'Liverpool', 'Sefton',
       'Knowsley', 'St. Helens', 'Manchester', 'Salford', 'Tameside',
       'Stockport', 'Bolton', 'Wigan', '

In [19]:
uk_accidents["District Area"].value_counts()

District Area
Birmingham            13491
Leeds                  8898
Manchester             6720
Bradford               6212
Sheffield              5710
                      ...  
Berwick-upon-Tweed      153
Teesdale                142
Shetland Islands        133
Orkney Islands          117
Clackmannanshire         91
Name: count, Length: 422, dtype: int64

<h3>Insight: The district area shows the 422 areas it has.</h3>
<hr>

<h2>03. Which weather condition is associated with the highest number of accidents?</h2>

In [20]:
uk_accidents["Weather_Conditions"].value_counts()

Weather_Conditions
Fine no high winds       535013
Raining no high winds     79696
Other                     17150
Raining + high winds       9615
Fine + high winds          8554
Snowing no high winds      6238
Fog or mist                3528
Snowing + high winds        885
Name: count, dtype: int64

<h3>Insight: The data shows that most accidents occurred during 'Fine no high winds' weather conditions, which had the highest count.</h3>

<hr>

<h2>04. Between urban and rural areas, where were the most casualties recorded?</h2>

In [21]:
areas_casualties = uk_accidents.groupby(['Urban_or_Rural_Area'])['Number_of_Casualties'].count()

In [22]:
areas_casualties

Urban_or_Rural_Area
Rural          238990
Unallocated        11
Urban          421678
Name: Number_of_Casualties, dtype: int64

<h3>Insight: <strong>Urban areas</strong> recorded the highest number of casualties from accidents, totaling 421,678, compared to 238,990 in rural areas. This suggests that accidents in urban settings lead to more casualties.</h3>

<hr>

<h2>05. Which of the three accident severity levels has the highest number of accidents?</h2>

In [23]:
uk_accidents["Accident_Severity"].unique()

array(['Serious', 'Slight', 'Fatal'], dtype=object)

In [24]:
serious_dataf = uk_accidents[uk_accidents['Accident_Severity'] == "Serious"]

In [25]:
serious_dataf

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious      5/6/2019  51.506187   
1       200701BS65737           Serious      2/7/2019  51.495029   
2       200701BS66127           Serious    26-08-2019  51.517715   
3       200701BS66128           Serious    16-08-2019  51.495478   
5       200701BS67159           Serious    18-09-2019  51.497750   
...               ...               ...           ...        ...   
660629  201091NL00686           Serious     11/2/2022  57.687933   
660645  201091NL05842           Serious    18-10-2022  60.162660   
660672  201091NM01085           Serious    29-01-2022  57.170725   
660677  201091NM01964           Serious    23-02-2022  57.214898   
660678  201091NM02142           Serious    28-02-2022  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
5                          Daylight  Kensington and Chelsea  -0.192556   
...                             ...                     ...        ...   
660629                     Daylight        Shetland Islands  -4.176860   
660645                     Daylight        Shetland Islands  -1.165232   
660672                     Daylight                Highland  -3.852421   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
5                          2                   3                     Dry   
...                      ...                 ...                     ...   
660629                     1                   2                     Dry   
660645                     1                   1                     Dry   
660672                     1                   2                    Snow   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2       Single carriageway               Urban     Fine no high winds   
3       Single carriageway               Urban     Fine no high winds   
5       Single carriageway               Urban     Fine no high winds   
...                    ...                 ...                    ...   
660629  Single carriageway               Rural      Fine + high winds   
660645  Single carriageway               Rural      Fine + high winds   
660672  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type  
0                                        Car  
1                                        Car  
2                      Taxi/Private hire car  
3       Bus or coach (17 or more pass seats)  
5                                        Car  
...                                      ...  
660629                                   Car  
660645                                   Car  
660672                                   Car  
660677                 Motorcycle over 500cc  
660

In [26]:
slight_dataf = uk_accidents[uk_accidents['Accident_Severity'] == "Slight"]

In [27]:
slight_dataf

Index Accident_Severity Accident Date   Latitude  \
4       200701BS66837            Slight      3/9/2019  51.488576   
8       200701BS67515            Slight    31-10-2019  51.493319   
9       200701BS67543            Slight    18-10-2019  51.484539   
13      200701BS67853            Slight     8/11/2019  51.508999   
14      200701BS67859            Slight     1/11/2019  51.485125   
...               ...               ...           ...        ...   
660671  201091NM01053            Slight    29-01-2022  57.572162   
660673  201091NM01190            Slight      2/2/2022  57.585152   
660674  201091NM01760            Slight    18-02-2022  57.374005   
660675  201091NM01881            Slight    21-02-2022  57.232273   
660676  201091NM01935            Slight    23-02-2022  57.585044   

              Light_Conditions           District Area  Longitude  \
4        Darkness - lights lit  Kensington and Chelsea  -0.192487   
8        Darkness - lights lit  Kensington and Chelsea  -0.173572   
9                     Daylight  Kensington and Chelsea  -0.175651   
13       Darkness - lights lit  Kensington and Chelsea  -0.210701   
14       Darkness - lights lit  Kensington and Chelsea  -0.167129   
...                        ...                     ...        ...   
660671                Daylight                Highland  -3.930799   
660673                Daylight                Highland  -3.748312   
660674                Daylight                Highland  -3.467828   
660675  Darkness - no lighting                Highland  -3.809281   
660676                Daylight                Highland  -3.862727   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
4                          1                   2                     Dry   
8                          1                   2                     Dry   
9                          1                   2                     Dry   
13                         3                   2             Wet or damp   
14                         1                   1                     Dry   
...                      ...                 ...                     ...   
660671                     1                   1            Frost or ice   
660673                     2                   2             Wet or damp   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   

                 Road_Type Urban_or_Rural_Area  Weather_Conditions  \
4       Single carriageway               Urban  Fine no high winds   
8       Single carriageway               Urban  Fine no high winds   
9       Single carriageway               Urban  Fine no high winds   
13      Single carriageway               Urban               Other   
14          One way street               Urban  Fine no high winds   
...                    ...                 ...                 ...   
660671  Single carriageway               Rural               Other   
660673  Single carriageway               Rural  Fine no high winds   
660674  Single carriageway               Rural  Fine no high winds   
660675  Single carriageway               Rural  Fine no high winds   
660676  Single carriageway               Rural  Fine no high winds   

                                 Vehicle_Type  
4                               Other vehicle  
8                                         Car  
9       Motorcycle over 125cc and up to 500cc  
13                                        Car  
14                                        Car  
...                                       ...  
660671    Van / Goods 3.5 tonnes mgw or under  
660673                                    Car  
660674                                    Car  
660675                                    Car  
660676                                    Car  

[563801 rows x 14 columns]

In [28]:
fatal_dataf = uk_accidents[uk_accidents['Accident_Severity'] == "Fatal"]

fatal_dataf

In [29]:
serious_dataf["Accident_Severity"].value_counts()

Accident_Severity
Serious    88217
Name: count, dtype: int64

In [30]:
slight_dataf["Accident_Severity"].value_counts()

Accident_Severity
Slight    563801
Name: count, dtype: int64

In [31]:
fatal_dataf["Accident_Severity"].value_counts()

Accident_Severity
Fatal    8661
Name: count, dtype: int64

<h3>Insight: Among the three levels of accident severity, 'Slight' accidents occur the most frequently, followed by 'Serious', while 'Fatal' accidents are the least common.</h3>

<hr>

<h2>06. Do the longitude and latitude coordinates of accidents show any correlation?</h2>

In [32]:
geographic_coor = uk_accidents['Latitude'].corr(uk_accidents['Longitude'])

In [33]:
geographic_coor

np.float64(-0.39811326552806847)

<h3>Insight: There is a moderate negative correlation (-0.40) between longitude and latitude, which probably reflects the general layout of where the accidents happened</h3>

<hr>

<h2>07. What Light Conditions has the highest number of occured accidents?</h2>

In [34]:
uk_accidents["Light_Conditions"].unique()

array(['Darkness - lights lit', 'Daylight', 'Darkness - lighting unknown',
       'Darkness - lights unlit', 'Darkness - no lighting'], dtype=object)

In [35]:
uk_accidents["Light_Conditions"].value_counts()

Light_Conditions
Daylight                       484880
Darkness - lights lit          129335
Darkness - no lighting          37437
Darkness - lighting unknown      6484
Darkness - lights unlit          2543
Name: count, dtype: int64

<h3>Insight: The data shows that in the Daylight time has the one who have the highest number of accidents that had happened, althroughout the whole record.</h3>

In [38]:
uk_accidents

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious      5/6/2019  51.506187   
1       200701BS65737           Serious      2/7/2019  51.495029   
2       200701BS66127           Serious    26-08-2019  51.517715   
3       200701BS66128           Serious    16-08-2019  51.495478   
4       200701BS66837            Slight      3/9/2019  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight    18-02-2022  57.374005   
660675  201091NM01881            Slight    21-02-2022  57.232273   
660676  201091NM01935            Slight    23-02-2022  57.585044   
660677  201091NM01964           Serious    23-02-2022  57.214898   
660678  201091NM02142           Serious    28-02-2022  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
4             Darkness - lights lit  Kensington and Chelsea  -0.192487   
...                             ...                     ...        ...   
660674                     Daylight                Highland  -3.467828   
660675       Darkness - no lighting                Highland  -3.809281   
660676                     Daylight                Highland  -3.862727   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2       Single carriageway               Urban     Fine no high winds   
3       Single carriageway               Urban     Fine no high winds   
4       Single carriageway               Urban     Fine no high winds   
...                    ...                 ...                    ...   
660674  Single carriageway               Rural     Fine no high winds   
660675  Single carriageway               Rural     Fine no high winds   
660676  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type  
0                                        Car  
1                                        Car  
2                      Taxi/Private hire car  
3       Bus or coach (17 or more pass seats)  
4                              Other vehicle  
...                                      ...  
660674                                   Car  
660675                                   Car  
660676                                   Car  
660677                 Motorcycle over 500cc  
660

<hr>
<h2>08. What type of a vehicle that often engaged from accidents?</h2>

In [41]:
uk_accidents["Vehicle_Type"].mode()

0    Car
Name: Vehicle_Type, dtype: object

<h3>Insight: The result clearly shows that the car got often engaged to an accidents.</h3>
<hr>

<h2>09. What road condition that the Urban area common have? </h2>

In [55]:
urban_area = uk_accidents[uk_accidents['Urban_or_Rural_Area'] == "Urban"]

In [56]:
urban_area

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious      5/6/2019  51.506187   
1       200701BS65737           Serious      2/7/2019  51.495029   
2       200701BS66127           Serious    26-08-2019  51.517715   
3       200701BS66128           Serious    16-08-2019  51.495478   
4       200701BS66837            Slight      3/9/2019  51.488576   
...               ...               ...           ...        ...   
660293  201063DP33610            Slight    29-10-2022  52.509366   
660309  201063DP35210            Slight     5/11/2022  52.513065   
660311  201063DP35410             Fatal    28-10-2022  52.515278   
660382  201063DP42910            Slight    21-12-2022  52.514222   
660383  201063DP43110            Slight    16-10-2022  52.520103   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
4             Darkness - lights lit  Kensington and Chelsea  -0.192487   
...                             ...                     ...        ...   
660293                     Daylight                   Powys  -3.331382   
660309                     Daylight                   Powys  -3.314105   
660311        Darkness - lights lit                   Powys  -3.317265   
660382                     Daylight                   Powys  -3.315171   
660383                     Daylight                   Powys  -3.303703   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660293                     1                   2                     Dry   
660309                     1                   2             Wet or damp   
660311                     1                   1                     Dry   
660382                     1                   1                    Snow   
660383                     1                   2                     Dry   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2       Single carriageway               Urban     Fine no high winds   
3       Single carriageway               Urban     Fine no high winds   
4       Single carriageway               Urban     Fine no high winds   
...                    ...                 ...                    ...   
660293  Single carriageway               Urban      Fine + high winds   
660309  Single carriageway               Urban                  Other   
660311  Single carriageway               Urban     Fine no high winds   
660382      One way street               Urban  Snowing no high winds   
660383  Single carriageway               Urban     Fine no high winds   

                                Vehicle_Type  
0                                        Car  
1                                        Car  
2                      Taxi/Private hire car  
3       Bus or coach (17 or more pass seats)  
4                              Other vehicle  
...                                      ...  
660293                                   Car  
660309                                   Car  
660311                         Other vehicle  
660382                                   Car  
660

In [57]:
uk_accidents["Road_Surface_Conditions"].unique()

array(['Dry', 'Wet or damp', 'Snow', 'Frost or ice',
       'Flood over 3cm. deep'], dtype=object)

In [58]:
urban_area = uk_accidents[uk_accidents['Urban_or_Rural_Area'] == "Urban"]["Road_Surface_Conditions"].mode()

In [59]:
urban_area

0    Dry
Name: Road_Surface_Conditions, dtype: object

<h3>Insight: In urban areas, the most common road surface condition is dry.</h3>
<hr>

<h2>10. Do accidents on wet roads happen more often during daylight or darkness? </h2>

<h2>11. Do single-carriageway roads have more severe accidents than other road types?</h2>

<h2>12. On average, how many vehicles are involved in fatal accidents compared to slight accidents? </h2>

<h2>13. Which district area reports the most accidents?</h2>

<h2>14. Which road type is most dangerous in terms of severe accidents?</h2>